src/transaction/log_global.c

```diff
@@ -131,12 +130,24 @@ log_global::initialize_log_prior_receiver ()
   m_prior_recver = std::make_unique<cublog::prior_recver> (prior_info);
 }
 
+void
+log_global::initialize_log_prior_sender ()
+{
+  m_prior_sender = std::make_unique<cublog::prior_sender> ();
+}
+
 void
 log_global::finalize_log_prior_receiver ()
 {
   m_prior_recver.reset (nullptr);
 }
 
+void
+log_global::finalize_log_prior_sender ()
+{
+  m_prior_sender.reset (nullptr);
```

Comment:
`prior_sender` must have a dtor which asserts that all sinks have been removed prior to destruction.
This will actually tell us if we have all un-init order right. On the happy path it will work. In edge cases, we'll see ... :)